### PR TITLE
Modularize API endpoints a bit more

### PR DESCRIPTION
**Pull request by valadaptive**
I've moved the files that register API endpoints into their own "endpoints" folder (maybe it should be called "routes" instead? I'm fine either way). Afterwards, I renamed some API endpoint paths to be more consistent, such that each file contains exactly one shared endpoint path.

After doing that, I refactored those endpoints to use Express's [Router API](https://expressjs.com/en/4x/api.html#router), so the app registers the routers itself. This should also allow for more modularization in the future, since specific middleware can be applied to specific routes.

The GitHub diff is a bit wonky here, so it might be best to review it one commit at a time with "Hide Whitespace" enabled.

### CODE REVIEW

File: public/scripts/chats.js [hunk 1]
```diff
@@ -152,7 +152,7 @@ export async function populateFileAttachment(message, inputId = 'file_form_input
  */
 export async function uploadFileAttachment(fileName, base64Data) {
     try {
-        const result = await fetch('/api/file/upload', {
+        const result = await fetch('/api/assets/upload', {
```
Comment (Cohee1207) [inline public/scripts/chats.js]: This endpoint name was not a mistake. It doesn't upload an asset.

File: public/scripts/extensions/stable-diffusion/index.js [hunk 2]
```diff
@@ -967,7 +967,7 @@ async function getAutoRemoteUpscalers() {
 
 async function getVladRemoteUpscalers() {
     try {
-        const result = await fetch('/api/sd-next/upscalers', {
+        const result = await fetch('/api/sd/next/upscalers', {
```
Comment (Cohee1207) [inline public/scripts/extensions/stable-diffusion/index.js]: sd-next is the name of the software, not a subspace of sd

https://github.com/vladmandic/automatic

File: public/scripts/chats.js (hunk 1 above)
Comment (valadaptive) [reply public/scripts/chats.js]: Is there a reason there's an API to upload files but not delete them? Regardless, I've moved it into a /files endpoint category.

File: public/scripts/extensions/stable-diffusion/index.js (hunk 2 above)
Comment (valadaptive) [reply public/scripts/extensions/stable-diffusion/index.js]: I've changed the endpoint path to `/api/sd/sd-next`, which matches the naming convention of `/api/sd/comfy` for ComfyUI

**Comment by Cohee1207** [comment]
@valadaptive 
It will take some time to review and test every moved endpoint to eliminate regression issues. Will be feasible to split this into smaller PRs? For example, one per module.

**Comment by valadaptive** [comment]
Sure; I'll try to.

**Comment by valadaptive** [comment]
I've split up the first few commits out into their own PRs. I'll split up the router-ification part after all the existing PRs are merged--if I have to make any changes, I don't want to have to rebase 17 different pull requests every time.

**Comment by valadaptive** [comment]
Closing in favor of new PRs to be made

**Pull request closed without merging**